pkg/kv/kvprober/kvprober.go

```diff
@@ -335,9 +368,21 @@ func (p *Prober) Start(ctx context.Context, stopper *stop.Stopper) error {
 					return
 				}
 
-				probeCtx, sp := tracing.EnsureChildSpan(ctx, p.tracer, opName+" - probe")
-				probe(probeCtx, pl)
-				sp.Finish()
+				func() {
+					// trace kv prober requests
+					probeCtx, collectAndFinish := tracing.ContextWithRecordingSpan(ctx, p.tracer, opName)
+
+					defer collectAndFinish()
+					if probe(probeCtx, pl) {
+						recording := collectAndFinish()
+
+						if traceDumpLimiter.ShouldLog() {
+							log.Health.Infof(ctx, "%v", recording)
```

Comment:
Arul suggests only logging the trace, if the `kvprober` probe fails, as per below:

> We might also want to consider just dumping the entire trace if a kvprober request fails in the logs

Want to try to implement that?

I think it will be wise to put the trace dumping in a second commit. So first commit just does leaseholder logging. Second commit does trace dumping. The SREs in India are good with `git`, so if not clear how to do this rebase, I am sure they can help. The reason to do two commits it to make easy to revert one of two, in case one has a problem. Also, it just makes for a clean git history, which is something CRL values.

Reply:
Yes, I'll look into implementing that and also have this trace dumping as a separate commit.